superset/views/core.py

```diff
@@ -1491,7 +1487,7 @@ def get_datasource_label(ds_name: utils.DatasourceName) -> str:
             tables = [tn for tn in tables if tn.schema in valid_schemas]
             views = [vn for vn in views if vn.schema in valid_schemas]
 
-        max_items = config.get("MAX_TABLE_NAMES") or len(tables)
+        max_items = config["MAX_TABLE_NAMES"] or len(tables)
```

Comment:
nit: isn't `..or tables` sufficient here?

Reply:
I think it's required because `max_items` is used as an index in a slice later on. I may be mistaken, and would love an education if so!

Reply:
Sorry, my bad!